typescript/integrations.d.ts

```diff
@@ -255,6 +256,21 @@ export type IntegrationGoogleMaps<S = SavedSecret> = IntegrationBase & {
   }
 }
 
+export type IntegrationSharepointEntraApplication<S = SavedSecret> = {
+  id: string
+  label: string
+  entraTenantId: string
+  entraApplicationId: string
+  entraClientSecret: ConstrainedSecret<S>
+}
+
+export type IntegrationSharepoint<S = SavedSecret> = IntegrationBase & {
+  type: 'SHAREPOINT'
+  configuration: {
```

Comment:
We going to need environments for this integration?

Reply:
we will be supporting multiple entra applications.